xcdat/averager.py

```diff
@@ -20,275 +23,311 @@
-    # If .bounds had a method to return a list of bounds entries, we could create weights for all dimensions
+# If .bounds had a method to return a list of bounds entries, we could create weights for all dimensions
 # May want to keep the axis weights or full weight array (i.e., add this information into the variable dataarray properties)
 # Is xvariable a good name?
 # Need to write unit tests
 # Can remove logging around bounds found in dataset
 # Functions are too specific to "lat" and "lon" (may have issues if dims are latitude / longitude)
 # Should remove these notes and outline...
 
+#: List of floats representing user-specified region axis bounds
+RegionAxisBounds = List[float]
+#: Dictionary of axis keys mapped to their bounds
+AxisWeights = Dict[Hashable, xr.DataArray]
+#: List of supported axis for spatial averaging
+SupportedAxes = Literal["lat", "lon"]
+SUPPORTED_AXES: Tuple[SupportedAxes, ...] = get_args(SupportedAxes)
 
-def _validate_region_lon_bounds(grid_lon_bounds, region_lon_bounds):
-    """Determines if values in a list of longitude coordinates falls within
-    longitudinal grid cell boundaries of a specified longitude axis.
 
-    Operations include:
+def _validate_region_lon_bounds(
+    region_bounds: RegionAxisBounds, domain_bounds: xr.DataArray
+) -> RegionAxisBounds:
+    """Check specified region lon bounds are within the domain lon bounds.
 
-    - Check to determine whether longitude values are within the boundaries of
-    the target longitude axis
+    If the region lon bounds are not in the domain lon bounds, try converting
+    the axis orientation of the region lon bounds (-180 to 180 or 0 to 360).
+    Afterwards, check if the converted region is in the domain again. If it still
+    isn't, this indicates that the region lon bounds are not correctly specified.
 
     Parameters
     ----------
-    grid_lon_bounds : xr.DataArray
-        longitude axis [len(lon), 2]
-    region_lon_bounds : float | int | list | array
-        list of longitude values to validate
+    region_bounds : RegionAxisBounds
+        Region longitude bounds.
+    domain_bounds : xr.DataArray
+        Domain longitude bounds.
 
     Returns
     -------
-    boolean
-        True if all input longitude bounds are within the boundaries of the
-        target longitudinal axis
+    RegionAxisBounds
+        Region longitude bounds.
 
-    Examples
-    --------
-    Import:
-    >>> from xcdat.dataset import open_dataset
-    >>> from xcdat.variable import open_variable
+    Raises
+    ------
+    ValueError
+        Specified region lon bounds are not in the domain lon bounds.
+    """
+    region_in_domain = _is_region_lon_in_domain(region_bounds, domain_bounds)
+    if not region_in_domain:
+        # TODO: Instead of the this implicit swapping of region axis,
+        # should we instead force the user to specify the correct lon bounds
+        # based on the axis orientation?
```

Comment:
# TODO: Instead of the this implicit swapping of region axis,
# should we instead force the user to specify the correct lon bounds
# based on the axis orientation?